blivet/iscsi.py

```diff
@@ -330,95 +324,64 @@ def discover(self, ipaddr, port="3260", username=None, password=None,
 
         Returns list of nodes user can log in.
         """
-        authinfo = None
 
         if not has_iscsi():
             raise IOError(_("iSCSI not available"))
         if self._initiator == "":
             raise ValueError(_("No initiator name set"))
 
-        if self.active_nodes((ipaddr, port)):
+        if self.active_nodes(TargetInfo(ipaddr, port)):
             log.debug("iSCSI: skipping discovery of %s:%s due to active nodes",
                       ipaddr, port)
         else:
-            if username or password or r_username or r_password:
-                # Note may raise a ValueError
-                authinfo = libiscsi.chapAuthInfo(username=username,
-                                                 password=password,
-                                                 reverse_username=r_username,
-                                                 reverse_password=r_password)
             self.startup()
-
-            # start libiscsi discover_sendtargets in a new process
-            # threads can't be used here because the libiscsi library
-            # using signals internally which are send to bad thread
-            (con_recv, con_write) = Pipe(False)
-            p = Process(target=_call_discover_targets, args=(con_write,
-                                                             con_recv,
-                                                             ipaddr,
-                                                             port,
-                                                             authinfo, ))
-            p.start()
-
-            # Close the writing end of the pipe in the parent
-            con_write.close()
-
-            with con_recv:
-                try:
-                    (ok, data) = con_recv.recv()
-                    if not ok:
-                        log.debug("iSCSI: exception raised when "
-                                  "discover_sendtargets process called: %s",
-                                  str(data))
```

Comment:
I want to ask. You removed this ugly error detection which is great but how this will behave when iscsi is for some reason throw an exception?